### PR TITLE
Add `v2` modules to integration tests

### PR DESCRIPTION
**Pull request by doriable**
This adds `v2` modules to all `buf/cmd/buf` workspace and lint integration tests.
It also covers a couple of a bit of clean-up around error messages.

### CODE REVIEW

**Review by bufdev — COMMENTED** [review]
I would love some tests added to probably `buflint_test.go` that test the new ignore_only paths being relative to the buf.yaml instead of being root-relative, and some tests that test excludes per directory in these integration tests, does that make sense/could you add those?

**Comment by bufdev** [comment]
Also, are there any tests checking how `v2` integrates with `--config`? I already can't remember the design decision I made regarding  `--config` with inline `v2` values, but would love us to lock in that behavior.

**Comment by doriable** [comment]
> Also, are there any tests checking how `v2` integrates with `--config`? I already can't remember the design decision I made regarding `--config` with inline `v2` values, but would love us to lock in that behavior.

So there currently is a test that inlines `v2` config with `--config`. It seems that the expectation right now is that it returns a non-zero status code (1) but does not provide a helpful error string.

https://github.com/bufbuild/buf/blob/4b7b26d54eab16ead054b42276fd4c24c4a128ee/private/buf/cmd/buf/workspace_test.go#L189-L207

File: private/bufpkg/bufcheck/buflint/buflint_test.go [hunk 1]
```diff
@@ -1109,30 +1168,40 @@ func testLintWithOptions(
 		storageos.ReadWriteBucketWithSymlinksIfSupported(),
 	)
 	require.NoError(t, err)
+	var options []bufworkspace.WorkspaceBucketOption
```
Comment (bufdev) [inline private/bufpkg/bufcheck/buflint/buflint_test.go:1171]: This isnt used

File: private/bufpkg/bufcheck/buflint/buflint_test.go [hunk 2]
```diff
@@ -1109,30 +1168,40 @@ func testLintWithOptions(
...
 		zap.NewNop(),
 		tracing.NopTracer,
 		readWriteBucket,
 		bufapi.NopClientProvider,
 		bufmodule.NopModuleDataProvider,
+		options...,
 	)
 	require.NoError(t, err)
 
+	if moduleFullNameString == "" {
```
Comment (bufdev) [inline private/bufpkg/bufcheck/buflint/buflint_test.go:1183]: change to:

```
opaqueID := moduleFullNameString
if opaqueID == "" {
  opaqueID = "."
}
```

File: private/bufpkg/bufmodule/module_set.go [hunk 3]
```diff
@@ -190,6 +190,40 @@ func ModuleSetToDAG(moduleSet ModuleSet) (*dag.Graph[string], error) {
 	return graph, nil
 }
 
+// ModuleToSelfContainedModuleReadBucketWithOnlyProtoFiles converts the Module to a
```
Comment (bufdev) [inline private/bufpkg/bufmodule/module_set.go:193]: Why did this move?

File: private/bufpkg/bufcheck/buflint/buflint_test.go (hunk 1 above)
Comment (doriable) [reply private/bufpkg/bufcheck/buflint/buflint_test.go]: Ah yes, I ended up going with targeting the opaque IDs instead

File: private/bufpkg/bufmodule/module_set.go (hunk 3 above)
Comment (doriable) [reply private/bufpkg/bufmodule/module_set.go]: Oh, I think I moved this by accident while traversing some of the helpers. Moved it back.

**Review by bufdev — APPROVED** [review]

**Pull request merged**